rocksdb_replicator/tests/rocksdb_replicator_test.cpp

```diff
@@ -623,6 +623,120 @@ TEST(RocksDBReplicatorTest, 1_master_1_slave_replication_mode_2) {
   EXPECT_EQ(100 /* FLAGS_replicator_timeout_ms */, replicated_db_master_shard1->current_replicator_timeout_ms_);
 }
 
+TEST(RocksDBReplicatorTest, 1_master_1_slave_1_observer_replication_mode_2) {
+  gflags::FlagSaver saver;
+
+  // Enable 2-ACK mode.
+  FLAGS_replicator_replication_mode = 2;
+  FLAGS_replicator_timeout_ms = 100;
+
+  // Setup a shard with 1 master,  1 slave, and 1 observer.
+  const int16_t master_port = 9092;
+  const int16_t slave_port = 9093;
+  const int16_t observer_port = 9094;
+  Host master(master_port);
+  Host slave(slave_port);
+  Host observer(observer_port);
+
+  auto db_master = cleanAndOpenDB("/tmp/db_master");
+  auto db_slave = cleanAndOpenDB("/tmp/db_slave");
+  auto db_observer = cleanAndOpenDB("/tmp/db_observer");
```

Comment:
shall we use leader/follower here to be consistent with the naming for ReplicaRole?

Reply:
yeah, I plan to change all places at the same time in a follow-up PR. This file still uses master/slave in other tests so it's okay to be consistent.